tests/robustness/multiclient_test/storagestats/storage_stats.go

```diff
@@ -0,0 +1,124 @@
+//go:build darwin || (linux && amd64)
+// +build darwin linux,amd64
+
+// Package storagestats contains logging mechanism
+// log disk space consumed by directories created by
+// robustness test framework before and after the test run.
+package storagestats
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/kopia/kopia/tests/robustness/multiclient_test/framework"
+)
+
+const (
+	logFileSubpath = "logs"
+)
+
+var logFilePath string
+
+// DirDetails represents details about a directory,
+// path, and size.
+type DirDetails struct {
+	DirPath string `json:"dirPath"`
+	DirSize int64  `json:"dirSize"`
+}
+
+// LogStorageStats logs disk space usage of provided dir paths.
+func LogStorageStats(ctx context.Context, dirs []string) error {
+	dd := collectDirDetails(dirs)
+
+	// write dir details into a JSON file
+	jsonData, err := json.Marshal(dd)
+	if err != nil {
+		return fmt.Errorf("error marshaling to JSON: %w", err)
+	}
+
+	file, err := createLogFile()
+	if err != nil {
+		return fmt.Errorf("error creating log file: %w", err)
+	}
+	defer func() error {
+		if err := file.Close(); err != nil {
+			return fmt.Errorf("error closing the log file: %w", err)
```

Comment:
The returned error is not checked, so this error is silently ignored.

Simply log the error. And change the function signature to not return anything.

(aside: there is separate technique to have the error be returned as part of `LogStorageStats`, but it does not seem that it is worth the trouble here.